src/Solver.cpp

```diff
@@ -725,6 +740,14 @@ void Solver::initializeFSRs() {
   _num_groups = _geometry->getNumEnergyGroups();
   _num_materials = _geometry->getNumMaterials();
 
+#ifdef NGROUPS
+  if (_geometry->getNumEnergyGroups() != _num_groups)
```

Comment:
_num_groups has been updated by L740, so this condition is always true. Please move L740 after this macro.